datastore/importers/nmdb-import-prowler/Parser.cpp

```diff
@@ -60,10 +64,14 @@ Parser::fromJsonV3(std::ifstream& _file)
   Data d;
   auto dataArray = json::parse(_file);
   for (const auto& entry : dataArray) {
-    LOG_DEBUG << "P1" << std::endl;
-    nmdop::ProwlerV3Data v3d {entry};
-    if (v3d != nmdop::ProwlerV3Data()) {
-      d.v3Data.push_back(v3d);
+    if (!entry.empty()) {
```

Comment:
Hey Marshall, what about in the case where the input file looks like: [{valid object},{}, {valid object}]. Wouldn't the middle element still be stored in `dataArray` as an empty Json object?

Reply:
I don't think so.  Line 69, below, should handle that case.  That is, line 68 will attempt to process the empty object but it's state will not change, so the test in line 69 will evaluate to `false`.

Reply:
After removing the line to check if the entry is empty, I realized that the check that's supposed to catch the empty objects, `if (v3d != nmdop::ProwlerV3Data())` will actually never catch an empty object because the assessmentStartTime's of v3d and the instantiated ProwlerV3Data will be off by a few milliseconds every time. Do you have any quick suggestion to get around that?

Reply:
Is that a guess or did you test and verify it (see 3)?  A few thoughts come to mind (no particular order).
1. Remove `assessmentStartTime` from the equality check.  Probably not the best answer, but the value shouldn't change in the middle of a tool run; so it should be a fairly static value in that case.
2. Alter or add an equality check that allows for an error bar.  You probably can use something like `boost::math::epsilon_difference` (see `IpNetwork.cpp` for a usage example).  Not sure if it's better to do it in the prowler related data objects or in the time one.
3. Modify the `Time` object's `readFormatted()` do a no-op if the time string it gets is empty.  However, I'm not sure the logic modifies the time outside of putting it in a specific format (which isn't part of the equality check).

There are probably more options, but not sure if it is needed.

Reply:
I'm wondering if we'd still need the if statement `if (v2d != nmdop::ProwlerV2Data()) {`

Reply:
Talked with you offline; still need it but the V2 needs to have the time check removed like what was done with V3.